### PR TITLE
propose change that will help short_address_function to work well

### PR DESCRIPTION
**Pull request by donaldte**

### CODE REVIEW

**Comment by tristanlatr** [comment]
Hi @donaldte,

Thanks for your contribution!

It looks like you've merge the changes from master to your develop branch before submitting this PR. 
Consequently, we should wait for the PR #114 to be merged before being able to merge this PR (we should merge it in the coming days). 

Otherwise you can cheery-pick commit https://github.com/LesFruitsDefendus/saskatoon-ng/pull/150/commits/ac6c0ebf11ba8c077f75d87b9d2de4d2888e1ac3 and apply it to a new clean branch from develop. 

It looks like your other PRs (#152, #153, #154) are in the same situation. It's not a big deal. 

Regarding the proposed changes, could you elaborate more on why disabling arguments `null` and `blank` from street `CharField` fix the `short_address_function()`. What is the current behaviour of `short_address_function()` ?and why is it problematic right now (If it fixes an issue that I didn't see, please link this issue so I can understand).
Actually, I don't see any `short_address_function()` in our codebase, so I'm a bot confused. I think your PR is overhaul missing a few explanations. Can you make more clear what your trying to accomplish ?

Thanks

Talk to you later,

**Comment by donaldte** [comment]
> Hi @donaldte,
> 
> Thanks for your contribution!
> 
> It looks like you've merge the changes from master to your develop branch before submitting this PR. Consequently, we should wait for the PR #114 to be merged before being able to merge this PR (we should merge it in the coming days).
> 
> Otherwise you can cheery-pick commit [ac6c0eb](https://github.com/LesFruitsDefendus/saskatoon-ng/commit/ac6c0ebf11ba8c077f75d87b9d2de4d2888e1ac3) and apply it to a new clean branch from develop and re-open a new PR.
> 
> It looks like your other PRs (#152, #153, #154) are in the same situation. It's not a big deal.
> 
> Regarding the proposed changes, could you elaborate more on why disabling arguments `null` and `blank` from street `CharField` fix the `short_address_function()`. What is the current behaviour of `short_address_function()` ?and why is it problematic right now (If it fixes an issue that I didn't see, please link this issue so I can understand). Actually, I don't see any `short_address_function()` in our codebase, so I'm a bot confused. I think your PR is overhaul missing a few explanations. Can you make more clear what your trying to accomplish ?
> 
> Thanks
> 
> Talk to you later,

Hi sir yes you are right for my differents PR they are pointing on master

but for the function that I'm talking about, here is the function:
def short_address(self):
        if self.street_number and self.street and self.complement:
            return "%s %s, %s" % (
                self.street_number,
                self.street,
                self.complement
            )
        elif self.street and self.street_number:
            return "%s %s" % (
                self.street_number,
                self.street
            )
        elif self.street and self.complement:
            return "%s, %s" % (
                self.street,
                self.complement
            )
        else:
            return self.street
if I call this function, this function need to return  the street name if the others arguments are empty but street name can be am empty string too because we have used blank=True. 

that is why I proposed to remove blank and null.

Thanks

**Comment by 2ynn** [comment]
@donaldte obviously we need to avoid having a property with all address fields blank but is there any exception raised if `self.street = None` is returned from `short_address()`? If so please indicate on which route. Thanks!

File: saskatoon/member/models.py [hunk 1]
```diff
@@ -289,8 +289,8 @@ class Organization(Actor):
     street = models.CharField(
         verbose_name=_("Street"),
         max_length=50,
-        null=True,
-        blank=True
+#         null=True,
+#         blank=True
```
Comment (2ynn) [inline saskatoon/member/models.py:293]: this will cause issues when migrating, if you remove null=True and blank=True, you need to provide a default

Comment (donaldte) [reply saskatoon/member/models.py:293]: Yes yes your are right sir it's true

**Comment by donaldte** [comment]
> @donaldte obviously we need to avoid having a property with all address fields blank but is there any exception raised if `self.street = None` is returned from `short_address()`? If so please indicate on which route. Thanks!

No sir it's okey I understand it now.  sorry

File: saskatoon/member/models.py (hunk 1 above)
Comment (2ynn) [reply saskatoon/member/models.py:293]: ok no worries. please don't use "sir" to refer to me. thanks!

Comment (2ynn) [reply saskatoon/member/models.py:293]: do you mind if we close this pull request?

Comment (donaldte) [reply saskatoon/member/models.py:293]: > do you mind if we close this pull request?

no @2ynn  you can close  thanks

**Pull request closed without merging**